shell.nix

```diff
@@ -27,12 +27,15 @@ with packageSet; haskell.packages.shellFor {
     dev.packages.hlint
     dev.packages.stylish-haskell
     dev.packages.haskell-language-server
+    dev.packages.hie-bios
     dev.packages.purty
     dev.packages.purs
     dev.packages.spago
     pkgs.stack
     dev.scripts.fixStylishHaskell
     dev.scripts.fixPurty
     dev.scripts.updateClientDeps
+    pkgs.rPackages.plotly
+    pkgs.python3
```

Comment:
What do we need these for? Worth trying to keep comments saying why things are in here otherwise it just gradually bloats and we can't remember when to remove things.